docs/pages/inboxes/push-notifs/understand-push-notifs.md

```diff
@@ -0,0 +1,141 @@
+- **Explain why your app doesn’t show a visible notification each time a push notification is received:** The server delivering notifications only knows of the existence of a conversation. It does not know who the sender or recipient is. That data is decoded on the device in the extension. As a result, it sends a push notification for every message that occurs in the conversation. We want to filter out notifications for notifications that the user themself sent.
+- **When your extension runs, what system and network resources does it need:** We might need to make a gRPC request to load additional information about a conversation. This is only necessary when we haven't stored the conversation details locally, which is expected to be less common than being able to just decrypt the conversation locally. For example, we might have a scenario in which a welcome message is pushed before it’s streamed.
+- **How often does your extension run? What can trigger it to run:** The extension will run whenever a message is sent or received in a conversation. The frequency will depend on how active a user is.
+
+## Understand displaying push notifications
+
+After the push notification service sends the notification, the app must [receive it](/inboxes/push-notifs/push-notifs#receive-push-notifications).
+
+There are some nuances to how push notifications can be handled once received by the app. While it is useful for all of these app types to use the XMTP push notification server filtering capabilities, it is especially important for an iOS app without the user notification entitlement.
+
+|  | Can decrypt before displaying? |
+| --- | --- |
+| Android and web apps | Yes |
+| iOS app with user notification entitlement | Yes |
+| iOS app without user notification entitlement | No |
+- If the app **can** **decrypt** the push notification before displaying it to the user, it can perform additional logic (should I display this?) before displaying the push notification. For example, the app can decrypt the push notification, see the topic type, and process it accordingly:
+    - Is the message in a welcome topic?
+        - `conversations.processWelcomeMessage`
+    - Is the message in a conversation topic?
+        - `conversation.processMessage`
+- If the app **cannot** **decrypt** the push notification before displaying it to the user, it can’t perform additional logic (should I display this?) before displaying the push notification.
+    
+    For example, without the user notification entitlement, the app cannot decrypt and modify the push notification before displaying it to the user. The notification arrives on the device, and iOS handles displaying it automatically. You can decrypt the content after the notification is shown, but you cannot intercept it before display and decide not to show it, for example.
+    
+
+## Understand HMAC keys and push notifications
+
```

Comment:
@codabrink - correct to leave out conversation membership and message authentication - and that HMAC keys are only used for Push Notifs?

Reply:
Yes, HMAC offers no security guarantees.